packages/trader/src/AppV2/Components/TradeParameters/RiskManagement/take-profit-and-stop-loss-container.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { observer } from 'mobx-react';
+import { useTraderStore } from 'Stores/useTraderStores';
+import { Button, useSnackbar } from '@deriv-com/quill-ui';
+import { localize, Localize } from '@deriv/translations';
+import { WS } from '@deriv/shared';
+import { getSnackBarText } from 'AppV2/Utils/trade-params-utils';
+import TakeProfitAndStopLossInput from './take-profit-and-stop-loss-input';
+
+type TTakeProfitAndStopLossContainerProps = {
+    closeActionSheet: () => void;
+    should_show_deal_cancellation?: boolean;
+};
+
+const TakeProfitAndStopLossContainer = observer(({ closeActionSheet }: TTakeProfitAndStopLossContainerProps) => {
+    const {
+        has_take_profit,
+        has_cancellation,
+        has_stop_loss,
+        take_profit,
+        onChangeMultiple,
+        stop_loss,
+        validation_errors,
+    } = useTraderStore();
+
+    const { addSnackbar } = useSnackbar();
+
+    const [tp_error_text, setTPErrorText] = React.useState(validation_errors?.take_profit?.[0] ?? '');
+    const tp_subscription_id_ref = React.useRef<string>();
+    const tp_ref = React.useRef({ has_take_profit, take_profit, tp_error_text: validation_errors?.take_profit?.[0] });
+    const is_be_response_tp_received_ref = React.useRef(false);
+
+    const [sl_error_text, setSLErrorText] = React.useState(validation_errors?.stop_loss?.[0] ?? '');
+    const sl_subscription_id_ref = React.useRef<string>();
+    const sl_ref = React.useRef({ has_stop_loss, stop_loss, sl_error_text: validation_errors?.stop_loss?.[0] });
+    const is_be_response_sl_received_ref = React.useRef(false);
+
+    const onSave = () => {
+        // Prevent from saving if user clicks before we got response from BE
+        if (!is_be_response_tp_received_ref.current && tp_ref.current.has_take_profit) return;
+        if (!is_be_response_sl_received_ref.current && sl_ref.current.has_stop_loss) return;
+
+        WS.forget(tp_subscription_id_ref.current);
```

Comment:
forgetting the subscription is already happening in the cleanup phase of `TakeProfitAndStopLossInput` why do need extra forget here?

Reply:
If Take profit is used separately (e.g. Accumulators, Turbos when property `has_save_button === true`) then clean up (`WS.forget`) will be done inside of `onSave` function in `TakeProfitAndStopLossInput`.
In case of Multipliers, `has_save_button === false`, so internal `onSave` function of `TakeProfitAndStopLossInput` won't be invoke. Instead, we will call `onSave` from `TakeProfitAndStopLossContainer` where e need to do the clean up